.github/workflows/gh-pages.yml

```
@@ -28,7 +32,7 @@ jobs:
           make -C docs html O="-t release"
 
       - name: Deploy
-        uses: peaceiris/actions-gh-pages@v3
+        uses: peaceiris/actions-gh-pages@373f7f263a76c20808c831209c920827a82a2847 # v3
```

A bit old, but should specify the full version.
```suggestion
        uses: peaceiris/actions-gh-pages@373f7f263a76c20808c831209c920827a82a2847 # v3.9.3
```